### ISSUE TITLE
Show trace ID in querier UI

### ISSUE THREAD

**Issue by fpetkovski** [thanos-io/thanos#5651]
### Is your proposal related to a problem?

When troubleshooting query performance through the querier, I would like be able to know what the trace ID for that query.

### Describe the solution you'd like

Would be nice to output the non-truncated trace ID in the UI, maybe next to the other PromQL related stats.
It would also be great to have the query as part of the tracing tags.

### Describe alternatives you've considered

The trace ID is currently only a response header, so I have to use the network inspector. It is also truncated for some reason so it's hard to look it up in a tracing system:
https://github.com/thanos-io/thanos/blob/319ef1500a5c626da0eee76f6e04a51a5afea2e6/pkg/tracing/jaeger/jaeger.go#L29-L34

**Comment by yeya24** [comment]
I really want to have the ability to be able to explore history queries in Thanos so that we can easily see slow queries and do analysis.
For example, something like https://docs.pingcap.com/tidb/stable/dashboard-slow-query would be super cool.
But I also understand that Thanos doesn't want to introduce dependencies like some relation databases. Perhaps we can start from an embedded sqlite?

**Comment by fpetkovski** [comment]
Hm I think that might be a bit out of the scope of this issue, I had in mind to at least start displaying trace IDs in the UI, and maybe add it to the slow query logs.

I think you might be able to achieve what you want by enabling slow query logs in query frontend?

**Comment by yeya24** [comment]
I think my idea covers the thing you proposed.
Basically, we want to have a way to store the metadata for each query somewhere and display the history queries from the UI. TraceID, query start time, end time and query stats, etc are all part of the query metadata.
I think using some persistence storage makes sense here like SQLite. Otherwise, If you want to display traceIDs how do you store them and the associated queries?

> I think you might be able to achieve what you want by enabling slow query logs in query frontend?
Yes, that's true. But I think we are talking about getting this feature in Thanos UI natively. If you can use logs then it is also easy to gather all the logs and extract all the traceIDs, too.

**Comment by fpetkovski** [comment]
For my use case, I don't need storage since I don't need to analyze past queries. I just want to know the trace id of the query that I just ran alongside other promql stats. The trace id is already sent back as a response header, so I just need that to show in the UI of the querier, and not be truncated.

**Comment by GiedriusS** [comment]
Yeah, it sounds like two different problems but @yeya24 your suggestion also sounds very good! Perhaps it could be a GSoC/LFX project?

**Comment by Vaani-pathariya** [comment]
I am new to thanos but I can contribute to this one !

**Comment by jatinagwal** [comment]
@fpetkovski I want to work on this issue. Just need your guidance, should I start working on this?

**Comment by fpetkovski** [comment]
Hey @jatinagwal, unless @Vaani-pathariya is already working on this, feel free to pick it up.

**Comment by jatinagwal** [comment]
Coolz. @Vaani-pathariya are you working on it ?

**Comment by Vaani-pathariya** [comment]
you may work on it @jatinagwal

**Comment by jatinagwal** [comment]
@fpetkovski , does it need a reactjs setup to show that ID in querier UI.

**Comment by fpetkovski** [comment]
It probably does, I believe the frontend code uses react.
